### PR TITLE
[deconz] Add On/Off thermostats

### LINKED ISSUE #14176: [deconz] Thermostat thing missing channel type: openclosesensor
I'm currently using Namron Panel Heater.
I have managed to get the thing created in openHAB.
But I'm missing the relay status.
This attribute is available in deConz and I can read it.
I assume this attribute is connect to channel type: open and item type: Contact
When I check the "documentation:
https://www.openhab.org/addons/bindings/deconz/
It seems like this channel is only available thru thing type: openclosesensor

Please add this channel also to the thing type Thermostate

### PR DESCRIPTION
**Pull request by J-N-K**
Closes #14176 

Also adds tests for correct channel addition and thing initialization for deconz.

### CODE REVIEW

**Review by phuongpham — APPROVED** [review]

File: bundles/org.openhab.binding.deconz/src/test/java/org/openhab/binding/deconz/internal/handler/SensorThermostatThingHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link SensorThermostatThingHandlerTest} is a
```
Comment (jlaur) [inline bundles/org.openhab.binding.deconz/src/test/java/org/openhab/binding/deconz/internal/handler/SensorThermostatThingHandlerTest.java:69]: Is a what? 🙂

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Can you please add the new channel type `on` to the README also, and regenerate the I18N properties file?

**Comment by J-N-K** [comment]
A good example why we should have more tests. I found some bugs due to the improved tests.

**Review by jlaur — CHANGES_REQUESTED** [review]
LGTM when I18N properties file is regenerated.

**Review by jlaur — APPROVED** [review]

**Comment by phuongpham** [comment]
@jlaur  I'm currently running on OH 3.4.2. Can you add support for 3.4.2 and not only 4.0
OH4.0 required gson 2.9 or higher

Import-Package: com.google.gson; version="[2.9.0,3.0.0)"

**Comment by jlaur** [comment]
> I'm currently running on OH 3.4.2. Can you add support for 3.4.2 and not only 4.0

Unfortunately I can't. This would require some work to backport since Java 17 features are used.

**Comment by phuongpham** [comment]
Ok I see. The thermostat status is not critical. So I can't use this new feature before I upgrade to OH 4 then.

**Pull request merged**